darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -1429,6 +1455,154 @@ def _load_from_checkpoint(self, file_path, **kwargs):
+        if file_name[:5] == "last-" or file_name[:5] == "best-":
+            checkpoint_dir = _get_checkpoint_folder(work_dir, model_name)
+        # manual save
+        else:
+            checkpoint_dir = os.getcwd()
+
+        ckpt_path = os.path.join(checkpoint_dir, file_name)
+        ckpt = torch.load(ckpt_path, **kwargs)
+        ckpt_hyper_params = ckpt["hyper_parameters"]
+
+        # verify that the arguments passed to the constructor match those of the checkpoint
+        for param_key, param_value in self.model_params.items():
+            # TODO: there are discrepancies between the param names, for ex num_layer/n_rnn_layers
+            if param_key in ckpt_hyper_params.keys() and param_value is not None:
+                # some parameters must be converted
+                if isinstance(ckpt_hyper_params[param_key], list) and not isinstance(
+                    param_value, list
+                ):
+                    param_value = [param_value] * len(ckpt_hyper_params[param_key])
+
+                raise_if(
+                    param_value != ckpt_hyper_params[param_key],
+                    f"The values of the hyper parameter {param_key} should be identical between "
+                    f"the instantiated model ({param_value}) and the loaded checkpoint "
+                    f"({ckpt_hyper_params[param_key]}). Please adjust the model accordingly.",
+                    logger,
+                )
+
+        # indicate to the user than checkpoints generated with darts <= 0.23.1 are not supported
+        raise_if_not(
+            "train_sample_shape" in ckpt.keys(),
+            "The provided checkpoint was generated with darts release <= 0.23.1"
+            " and it is missing the 'train_sample_shape' key. This value must"
+            " be computed from the `model.train_sample` attribute and manually"
+            " added to the checkpoint prior to loading.",
+            logger,
+        )
+
+        # pl_forecasting module saves the train_sample shape, must recreate one
+        mock_train_sample = [
+            np.zeros(sample_shape) if sample_shape else None
+            for sample_shape in ckpt["train_sample_shape"]
+        ]
+        self.train_sample = tuple(mock_train_sample)
+
+        # instanciate the model without having to call `fit_from_dataset`
+        self._init_model()
+        # cast model precision to correct type
+        self.model.to_dtype(ckpt["model_dtype"])
+        # load only the weights from the state dict
+        self.model.load_state_dict(ckpt["state_dict"], strict=strict)
+        # update the fit_called attribute to allow for direct inference
+        self._fit_called = True
+
+    def load_weights(self, path: str, **kwargs):
+        """
+        Loads a model weights from a Pytorch Lightning checkpoint file ('.ckpt') path.
+
+        This method is used for the checkpoints created with ``model.save()``
```

Comment:
(see my comment below)
```suggestion
        Loads the weights from a manually saved model (saved with :meth:`save() <TorchForecastingModel.save()>`).
```